api/post.go

```diff
@@ -248,7 +251,24 @@ func handlePostEventsAndForget(c *Context, post *model.Post, triggerWebhooks boo
 			channel = result.Data.(*model.Channel)
 		}
 
-		sendNotificationsAndForget(c, post, team, channel)
+		var profiles map[string]*model.User
+		if result := <-pchan; result.Err != nil {
+			l4g.Error(utils.T("api.post.handle_post_events_and_forget.profiles.error"), c.Session.TeamId, result.Err)
+			return
+		} else {
+			profiles = result.Data.(map[string]*model.User)
+		}
+
+		var members []model.ChannelMember
+		if result := <-mchan; result.Err != nil {
+			l4g.Error(utils.T("api.post.handle_post_events_and_forget.members.error"), post.ChannelId, result.Err)
+			return
+		} else {
+			members = result.Data.([]model.ChannelMember)
+		}
+
+		sendNotificationsAndForget(c, post, team, channel, profiles, members)
+		go checkForOutOfChannelMentions(c, post, channel, profiles, members)
```

Comment:
I thought it might be better to transition the asynchronous calls towards being explicitly called with go instead of doing it inside the function and adding AndForget to the end of the name. That way, we can run tests and such on the synchronous version easily. I'm 1/5 that we should do it this way, but I wasn't going to bother with the existing function since it would just mess up the indenting, but seeing as I've already changed the indenting, I can do it now.

@coreyhulen Does that sound good with you?